[adaptive_]max_pool{1,2,3}d: handle edge case when input is filled with -inf

Baranowski opened the pull request:
Fixes #40131

Baranowski committed:
REVERTME MWE

Baranowski committed:
adaptive_maxpool_{2,3}d: maxindex=0 instead of -1

Baranowski committed:
Assign -inf by default

Baranowski committed:
REVERTME debugging

Baranowski committed:
Revert "REVERTME debugging"

This reverts commit 3b08ca8f6ddbddefab0088d135c54bb29f25a130.

Baranowski committed:
Revert "REVERTME MWE"

This reverts commit 49d166cced57fe048138e57a3612d2a02288d61c.

Baranowski committed:
FractionalMaxPool{2,3}d

Baranowski committed:
REVERTME Revert C++ changes for FractionalMaxPool{2,3}d

Baranowski committed:
Revert "REVERTME Revert C++ changes for FractionalMaxPool{2,3}d"

This reverts commit 302276ced0a8b2f05fc12c7dde8b1ee642088b8f.

Baranowski commented:
@xwang233, I didn't know who would be appropriate so I have marked you as a reviewer.

I will create the XLA issue after the first review.

xwang233 reviewed (commented):
Thanks for the fix! Overall it looks good. Only the default max index needs to be changed.

xwang233 commented:
I'm not familiar with the XLA test. If XLA failure is real, we can temporarily disable test on XLA and leave a TODO there. Seems like `@onlyOnCPUAndCUDA` does the work.

Baranowski committed:
onlyOnCPUAndCUDA

Baranowski committed:
Proper initial values for maxIndex

Baranowski commented:
Good call, @xwang233. Fixed

Baranowski committed:
Lint

xwang233 commented on aten/src/ATen/native/AdaptiveMaxPooling2d.cpp:
```diff
@@ -59,12 +59,12 @@ static void adaptive_max_pool2d_single_out_frame(
+          for(; ih < kH; ih++)
           {
-            for(iw = 0; iw < kW; iw++)
+            for(; iw < kW; iw++)
```
I think the initialization of `iw=0` shouldn't be removed. Same below.

Baranowski committed:
Initialize loop variables

Baranowski commented:
Thanks for the quick and patient reviews, @xwang233. I won't have the time to work on this after Sunday next week (July 19th) so it would be brilliant to get this PR merged before then.

xwang233 commented:
@Baranowski Thanks. I don't work at FB, so I can't merge this for you.

ezyang commented:
It would be good to get a performance comparison before and after here.

Baranowski committed:
Benchmarks

Baranowski commented:
I have added benchmarks in the most recent commit. I don't see significant differences.

Master:

<details>

```
(pytorch-cuda-dev) wbaranowski@qgpu2:~/git/Quansight/vanilla/benchmarks/operator_benchmark$ python -m pt.pool_test --tag all
# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,1]_stride[2,1]_N1_C16_H32_W32_cpu
# Input: kernel: [3, 1], stride: [2, 1], N: 1, C: 16, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 43.277

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,2]_stride[2,2]_N8_C32_H32_W32_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 8, C: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 74.007

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,2]_stride[2,2]_N8_C32_H32_W64_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 8, C: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 103.433

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,2]_stride[2,2]_N8_C32_H64_W32_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 8, C: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 101.649

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,2]_stride[2,2]_N8_C32_H64_W64_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 8, C: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 158.317

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,2]_stride[2,2]_N16_C32_H32_W32_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 16, C: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 77.478

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,2]_stride[2,2]_N16_C32_H32_W64_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 16, C: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 116.714

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,2]_stride[2,2]_N16_C32_H64_W32_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 16, C: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 116.661

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,2]_stride[2,2]_N16_C32_H64_W64_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 16, C: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 158.889

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,3]_stride[2,2]_N8_C32_H32_W32_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 8, C: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 83.278

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,3]_stride[2,2]_N8_C32_H32_W64_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 8, C: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 126.931

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,3]_stride[2,2]_N8_C32_H64_W32_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 8, C: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 113.920

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,3]_stride[2,2]_N8_C32_H64_W64_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 8, C: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 182.647

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,3]_stride[2,2]_N16_C32_H32_W32_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 16, C: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 98.402

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,3]_stride[2,2]_N16_C32_H32_W64_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 16, C: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 111.891

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,3]_stride[2,2]_N16_C32_H64_W32_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 16, C: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 117.598

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,3]_stride[2,2]_N16_C32_H64_W64_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 16, C: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 241.957

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,1]_stride[2,1]_N1_C16_H32_W32_cpu
# Input: kernel: [3, 1], stride: [2, 1], N: 1, C: 16, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 39.386

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,2]_stride[2,2]_N8_C32_H32_W32_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 8, C: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 54.870

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,2]_stride[2,2]_N8_C32_H32_W64_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 8, C: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 54.467

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,2]_stride[2,2]_N8_C32_H64_W32_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 8, C: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 54.567

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,2]_stride[2,2]_N8_C32_H64_W64_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 8, C: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 54.727

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,2]_stride[2,2]_N16_C32_H32_W32_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 16, C: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 64.878

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,2]_stride[2,2]_N16_C32_H32_W64_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 16, C: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 63.164

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,2]_stride[2,2]_N16_C32_H64_W32_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 16, C: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 63.417

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,2]_stride[2,2]_N16_C32_H64_W64_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 16, C: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 62.819

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,3]_stride[2,2]_N8_C32_H32_W32_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 8, C: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 55.215

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,3]_stride[2,2]_N8_C32_H32_W64_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 8, C: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 54.947

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,3]_stride[2,2]_N8_C32_H64_W32_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 8, C: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 55.018

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,3]_stride[2,2]_N8_C32_H64_W64_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 8, C: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 58.489

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,3]_stride[2,2]_N16_C32_H32_W32_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 16, C: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 67.708

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,3]_stride[2,2]_N16_C32_H32_W64_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 16, C: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 64.613

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,3]_stride[2,2]_N16_C32_H64_W32_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 16, C: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 64.880

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,3]_stride[2,2]_N16_C32_H64_W64_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 16, C: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 65.123

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,1,3]_stride[2,1,2]_N1_C16_D16_H32_W32_cpu
# Input: kernel: [3, 1, 3], stride: [2, 1, 2], N: 1, C: 16, D: 16, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 324.005

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N8_C32_D32_H32_W32_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 2255.940

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N8_C32_D32_H32_W64_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 4465.090

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N8_C32_D32_H64_W32_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 4151.426

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N8_C32_D32_H64_W64_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 10130.852

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N16_C32_D32_H32_W32_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 2403.098

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N16_C32_D32_H32_W64_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 7265.930

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N16_C32_D32_H64_W32_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 6345.975

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N16_C32_D32_H64_W64_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 13842.293

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N8_C32_D32_H32_W32_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 1804.096

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N8_C32_D32_H32_W64_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 4145.099

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N8_C32_D32_H64_W32_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 3979.806

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N8_C32_D32_H64_W64_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 10293.466

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N16_C32_D32_H32_W32_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 2501.046

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N16_C32_D32_H32_W64_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 7369.280

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N16_C32_D32_H64_W32_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 6874.013

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N16_C32_D32_H64_W64_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 14725.793

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,1,3]_stride[2,1,2]_N1_C16_D16_H32_W32_cpu
# Input: kernel: [3, 1, 3], stride: [2, 1, 2], N: 1, C: 16, D: 16, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 39.616

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N8_C32_D32_H32_W32_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 78.150

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N8_C32_D32_H32_W64_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 74.670

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N8_C32_D32_H64_W32_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 71.217

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N8_C32_D32_H64_W64_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 77.309

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N16_C32_D32_H32_W32_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 88.279

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N16_C32_D32_H32_W64_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 85.710

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N16_C32_D32_H64_W32_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 87.579

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N16_C32_D32_H64_W64_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 139.081

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N8_C32_D32_H32_W32_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 74.413

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N8_C32_D32_H32_W64_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 83.784

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N8_C32_D32_H64_W32_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 77.414

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N8_C32_D32_H64_W64_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 100.437

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N16_C32_D32_H32_W32_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 90.312

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N16_C32_D32_H32_W64_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 94.083

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N16_C32_D32_H64_W32_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 93.086

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N16_C32_D32_H64_W64_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 145.713

(pytorch-cuda-dev) wbaranowski@qgpu2:~/git/Quansight/vanilla/benchmarks/operator_benchmark$
```

</details>

Modified version:

<details>

```
(pytorch-cuda-dev) wbaranowski@qgpu2:~/git/Quansight/pytorch/benchmarks/operator_benchmark$ python -m pt.
pool_test --tag all
# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,1]_stride[2,1]_N1_C16_H32_W32_cpu
# Input: kernel: [3, 1], stride: [2, 1], N: 1, C: 16, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 42.339

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,2]_stride[2,2]_N8_C32_H32_W32_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 8, C: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 74.987

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,2]_stride[2,2]_N8_C32_H32_W64_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 8, C: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 103.705

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,2]_stride[2,2]_N8_C32_H64_W32_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 8, C: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 102.937

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,2]_stride[2,2]_N8_C32_H64_W64_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 8, C: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 178.586

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,2]_stride[2,2]_N16_C32_H32_W32_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 16, C: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 94.048

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,2]_stride[2,2]_N16_C32_H32_W64_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 16, C: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 106.972

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,2]_stride[2,2]_N16_C32_H64_W32_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 16, C: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 112.960

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,2]_stride[2,2]_N16_C32_H64_W64_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 16, C: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 159.002

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,3]_stride[2,2]_N8_C32_H32_W32_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 8, C: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 83.364

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,3]_stride[2,2]_N8_C32_H32_W64_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 8, C: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 110.988

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,3]_stride[2,2]_N8_C32_H64_W32_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 8, C: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 115.071

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,3]_stride[2,2]_N8_C32_H64_W64_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 8, C: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 169.171

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,3]_stride[2,2]_N16_C32_H32_W32_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 16, C: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 83.448

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,3]_stride[2,2]_N16_C32_H32_W64_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 16, C: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 112.358

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,3]_stride[2,2]_N16_C32_H64_W32_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 16, C: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 114.631

# Benchmarking PyTorch: AdaptiveMaxPool2d
# Mode: Eager
# Name: AdaptiveMaxPool2d_kernel[3,3]_stride[2,2]_N16_C32_H64_W64_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 16, C: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 168.563

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,1]_stride[2,1]_N1_C16_H32_W32_cpu
# Input: kernel: [3, 1], stride: [2, 1], N: 1, C: 16, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 38.105

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,2]_stride[2,2]_N8_C32_H32_W32_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 8, C: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 53.678

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,2]_stride[2,2]_N8_C32_H32_W64_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 8, C: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 58.790

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,2]_stride[2,2]_N8_C32_H64_W32_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 8, C: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 52.641

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,2]_stride[2,2]_N8_C32_H64_W64_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 8, C: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 54.202

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,2]_stride[2,2]_N16_C32_H32_W32_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 16, C: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 64.891

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,2]_stride[2,2]_N16_C32_H32_W64_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 16, C: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 61.271

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,2]_stride[2,2]_N16_C32_H64_W32_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 16, C: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 62.074

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,2]_stride[2,2]_N16_C32_H64_W64_cpu
# Input: kernel: [3, 2], stride: [2, 2], N: 16, C: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 62.022

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,3]_stride[2,2]_N8_C32_H32_W32_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 8, C: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 55.512

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,3]_stride[2,2]_N8_C32_H32_W64_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 8, C: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 56.149

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,3]_stride[2,2]_N8_C32_H64_W32_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 8, C: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 59.689

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,3]_stride[2,2]_N8_C32_H64_W64_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 8, C: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 59.210

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,3]_stride[2,2]_N16_C32_H32_W32_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 16, C: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 69.061

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,3]_stride[2,2]_N16_C32_H32_W64_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 16, C: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 67.673

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,3]_stride[2,2]_N16_C32_H64_W32_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 16, C: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 67.671

# Benchmarking PyTorch: FractionalMaxPool2d
# Mode: Eager
# Name: FractionalMaxPool2d_kernel[3,3]_stride[2,2]_N16_C32_H64_W64_cpu
# Input: kernel: [3, 3], stride: [2, 2], N: 16, C: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 67.627

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,1,3]_stride[2,1,2]_N1_C16_D16_H32_W32_cpu
# Input: kernel: [3, 1, 3], stride: [2, 1, 2], N: 1, C: 16, D: 16, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 338.750

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N8_C32_D32_H32_W32_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 1580.897

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N8_C32_D32_H32_W64_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 4679.794

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N8_C32_D32_H64_W32_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 3772.246

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N8_C32_D32_H64_W64_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 9960.207

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N16_C32_D32_H32_W32_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 2765.616

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N16_C32_D32_H32_W64_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 6874.791

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N16_C32_D32_H64_W32_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 6816.528

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N16_C32_D32_H64_W64_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 14006.464

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N8_C32_D32_H32_W32_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 1693.150

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N8_C32_D32_H32_W64_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 4250.506

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N8_C32_D32_H64_W32_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 4019.097

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N8_C32_D32_H64_W64_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 9947.090

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N16_C32_D32_H32_W32_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 2356.209

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N16_C32_D32_H32_W64_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 7095.732

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N16_C32_D32_H64_W32_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 6791.053

# Benchmarking PyTorch: AdaptiveMaxPool3d
# Mode: Eager
# Name: AdaptiveMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N16_C32_D32_H64_W64_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 14499.389

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,1,3]_stride[2,1,2]_N1_C16_D16_H32_W32_cpu
# Input: kernel: [3, 1, 3], stride: [2, 1, 2], N: 1, C: 16, D: 16, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 40.421

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N8_C32_D32_H32_W32_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 73.324

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N8_C32_D32_H32_W64_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 76.495

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N8_C32_D32_H64_W32_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 75.037

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N8_C32_D32_H64_W64_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 76.399

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N16_C32_D32_H32_W32_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 104.319

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N16_C32_D32_H32_W64_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 98.106

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N16_C32_D32_H64_W32_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 96.371

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,2,3]_stride[2,2,2]_N16_C32_D32_H64_W64_cpu
# Input: kernel: [3, 2, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 96.063

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N8_C32_D32_H32_W32_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 80.156

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N8_C32_D32_H32_W64_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 81.476

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N8_C32_D32_H64_W32_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 81.561

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N8_C32_D32_H64_W64_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 8, C: 32, D: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 81.320

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N16_C32_D32_H32_W32_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 32, W: 32, device: cpu
Forward Execution Time (us) : 100.050

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N16_C32_D32_H32_W64_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 32, W: 64, device: cpu
Forward Execution Time (us) : 96.316

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N16_C32_D32_H64_W32_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 64, W: 32, device: cpu
Forward Execution Time (us) : 93.940

# Benchmarking PyTorch: FractionalMaxPool3d
# Mode: Eager
# Name: FractionalMaxPool3d_kernel[3,3,3]_stride[2,2,2]_N16_C32_D32_H64_W64_cpu
# Input: kernel: [3, 3, 3], stride: [2, 2, 2], N: 16, C: 32, D: 32, H: 64, W: 64, device: cpu
Forward Execution Time (us) : 94.695

(pytorch-cuda-dev) wbaranowski@qgpu2:~/git/Quansight/pytorch/benchmarks/operator_benchmark$
```

</details>
